packages/components/src/components/menu/menu.component.ts

```diff
@@ -33,10 +36,27 @@ export default class SynMenu extends SynergyElement {
   static styles: CSSResultGroup = [componentStyles, styles, customStyles];
 
   @query('slot') defaultSlot: HTMLSlotElement;
+  @state() hasMenuItemsWithCheckmarks = false;
+
+  private handleUpdateCheckmarks(items: SynMenuItem[]) {
+    // #368: Treat a menu as having checkmarks if it has any checkboxes or items with loading states
+    // The loading indicator has to be checked as well, as it's specially placed over the check mark
+    this.hasMenuItemsWithCheckmarks = items.some(item => item.type === 'checkbox' || item.loading);  
+  }
+
+  private updateCheckMarksByChildPropChange = (e: SynAttributesChangedEvent) => {
+    e.stopPropagation();
```

Comment:
Currently if adding an EventListener on the syn-menu component itself the `syn-attributes-changed` from the menu-item can be catched. 
E.g. following code results in both console.logs are done when changing the syn-menu-item to e.g. loading=true: 

```js
    <syn-menu style="max-width: 240px;">
      <syn-menu-item>
        <syn-icon slot="prefix" name="house"></syn-icon>
        Home
      </syn-menu-item>
    </syn-menu>
    <script type="module">
      const menu = document.querySelector('syn-menu');
      menu.addEventListener('syn-attributes-changed', (event) => {
        console.log('##### syn-menu');
      });
      const menuItem = document.querySelector('syn-menu-item');
      menuItem.addEventListener('syn-attributes-changed', (event) => {
        console.log('##### syn-menu-item');
      });
    </script>
```

If we don't want the event to bubble this far, it seems we need to do a `e.stopImmediatePropagation();`.
What do you mean? Maybe it even doesn't matter, but I was just curious what happens if I listen on the syn-menu for it

Reply:
No you are right, as this is an implementation detail, it should not process any further.

Reply:
Fixed in latest version